scripts/py_matter_yamltests/matter_yamltests/parser.py

```diff
@@ -710,6 +725,14 @@ def post_process_response(self, received_responses):
         if self._test.save_response_as:
             self._runtime_config_variable_storage[self._test.save_response_as] = received_responses
 
+        if self.is_event:
+            last_event_number = self._get_last_event_number(received_responses)
+            if last_event_number:
+                if 'LastReceivedEventNumber' not in self._runtime_config_variable_storage:
+                    self._runtime_config_variable_storage['LastReceivedEventNumber'] = last_event_number
+                elif self._runtime_config_variable_storage['LastReceivedEventNumber'] < last_event_number:
+                    self._runtime_config_variable_storage['LastReceivedEventNumber'] = last_event_number
```

Comment:
Can't we just `self._runtime_config_variable_storage['LastReceivedEventNumber'] = last_event_number` instead of checking if it is not defined and/or if `last_event_number` is superior to the `LastReceivedEventNumber` ?

If `last_event_number` is less than `LastReceivedEventNumber` what should happens ? Should we raise an error ?

Reply:
I maintained logic from previous code, however I am indeed unclear why we would receive "old" events before new ones.

Reply:
We can raise ... I will add a line for now since I am unclear when/if this happens. Should not happen but would rather not introduce new throws in the code.